parsec/core/fs/storage/local_database.py

```diff
@@ -127,7 +86,7 @@ def _manage_operational_error(self, allow_commit: bool = False) -> Iterator[None
 
             # Close the sqlite3 connection
             try:
-                self._conn.close()
+                await self.run_in_thread(self._conn.close)
```

Comment:
Just to be sure: given the call is now async, it is impacted by `trio.Cancelled` exception
It should have an impact for the main case (i.e. when an `OperationalError` is raised), but in more complex case (`OperationalError` then `trio.Cancelled`) the behavior is different

Reply:
4ba56f6623e2f17bea1d06876885307bfb6c61b4